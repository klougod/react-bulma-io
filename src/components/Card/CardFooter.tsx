import React, { ReactNode } from 'react'

interface CardFooterProps {
  className?: string,
  children?: ReactNode
}

export const CardFooter = ({ className, children }: CardFooterProps) => (
  <header className={`card-footer ${className}`}>{children}</header>
)

CardFooter.defaultProps = {
  className: ''
}

export default CardFooter
