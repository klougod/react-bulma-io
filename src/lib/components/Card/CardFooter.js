import React from 'react'
import PropTypes from 'prop-types'

export const CardFooter = ({ className, children }) => (
  <header className={`card-footer ${className}`}>
    { children }
  </header>
)

CardFooter.defaultProps = {
  className: ''
}

CardFooter.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardFooter