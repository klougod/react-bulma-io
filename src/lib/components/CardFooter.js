import React from 'react'
import PropTypes from 'prop-types'

export const CardFooter = props => {

  const {
    className,
    children
  } = props

  return (
    <header className={`card-footer ${className}`}>
      { children }
    </header>
  )
}

CardFooter.defaultProps = {
  className: ''
}

CardFooter.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
}

export default CardFooter