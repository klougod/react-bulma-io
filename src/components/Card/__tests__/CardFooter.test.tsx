import React from 'react'
import Enzyme from 'enzyme'

import CardFooter from '../CardFooter'

describe('CardFooter component test', function () {
  it('should render without crash without props', function () {
    const cardFooterMock = Enzyme.shallow(<CardFooter />)
    expect(cardFooterMock).toMatchSnapshot()
  })

  it('should render without crash with props', function () {
    const cardFooterMockProps = {
      className: 'cardFooter CLass'
    }
    const cardFooterMock = Enzyme.shallow(<CardFooter {...cardFooterMockProps} />)
    expect(cardFooterMock).toMatchSnapshot()
  })
})
