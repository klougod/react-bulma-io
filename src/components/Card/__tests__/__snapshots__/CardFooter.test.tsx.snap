// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooter component test should render without crash with props 1`] = `
<header
  className="card-footer cardFooter CLass"
/>
`;

exports[`CardFooter component test should render without crash without props 1`] = `
<header
  className="card-footer "
/>
`;
